app/aggregates/deciding/decision.rb

```diff
@@ -110,21 +121,23 @@ def send_to_provider(user_id:, application_id:)
 
     on Unlinked do |_event|
       @application_id = nil
-      @reference = nil
       @comment = nil
     end
 
     on Linked do |event|
       @application_id = event.data.fetch(:application_id)
-      @reference = event.data.fetch(:reference, nil)
     end
 
-    # Original reference is retained when linking to CIFC
     on LinkedToCifc do |event|
       @application_id = event.data.fetch(:application_id)
       @state = Types::DecisionState[:draft]
     end
 
+    on LinkedToNafi do |event|
+      @application_id = event.data.fetch(:application_id)
+      @state = Types::DecisionState[:draft]
+    end
+
```

Comment:
Although it currently behaves the same way as LinkedToCifc, we record it as a different event type to preserve the history of a linking.